issuetracker/src/main/java/org/openjdk/skara/issuetracker/jira/JiraIssue.java

```diff
@@ -291,9 +298,13 @@ public void setLabels(List<String> labels) {
 
     @Override
     public List<Label> labels() {
-        return json.get("fields").get("labels").stream()
-                   .map(s -> new Label(s.asString()))
-                   .collect(Collectors.toList());
+        if(labels == null){
+            json = request.get("").execute();
+            labels = json.get("fields").get("labels").stream()
+                    .map(s -> new Label(s.asString()))
+                    .collect(Collectors.toList());
+        }
```

Comment:
I tried to just update the labels but I could not make it. So I update the json first and then update the labels.